### PR TITLE
spread: add boilerplate for Linode delta uploads

### PR DESCRIPTION
**Pull request by niemeyer**

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -90,6 +92,10 @@ path: /home/gopath/src/github.com/snapcore/snapd
 
 exclude:
     - .git
+    - cmd/snap/snap
```
Comment (zyga) [inline spread.yaml]: It would be nice to be able to tell spread to ignore any files ignored by git already (via .gitignore)

File: spread.yaml [hunk 2]
```diff
@@ -101,8 +107,40 @@ debug-each: |
     journalctl -u snapd
     dmesg | grep DENIED
 
+rename:
+    # Move content into a directory, so that deltas computed by repack benefit
+    # from the content looking similar to codeload.github.com.
+    - s,^,$DELTA_PREFIX,S
```
Comment (zyga) [inline spread.yaml]: What is `S` there?

File: spread.yaml [hunk 3]
```diff
@@ -101,8 +107,40 @@ debug-each: |
...
+
+repack: |
+    # For Linode, compute a delta based on a known git reference that can be
+    # obtained directly from GitHub. There's nothing special about that reference,
+    # other than it will often be in the local repository's history already.
+    # The more recent the reference, the smaller the delta.
+    if ! echo $SPREAD_BACKENDS | grep linode; then
```
Comment (zyga) [inline spread.yaml]: `$SPREAD_BACKEND` or `$SPREAD_BACKENDS`?

File: spread.yaml [hunk 4]
```diff
@@ -101,8 +107,40 @@ debug-each: |
...
+        cat <&3 >&4
```
Comment (zyga) [inline spread.yaml]: Can you please document what is on those two descriptors?

File: spread.yaml (hunk 1 above)
Comment (niemeyer) [inline spread.yaml]: Indeed.

File: spread.yaml (hunk 2 above)
Comment (niemeyer) [inline spread.yaml]: https://github.com/snapcore/spread#including

File: spread.yaml (hunk 3 above)
Comment (niemeyer) [inline spread.yaml]: $SPREAD_BACKEND holds the currently running backend.
$SPREAD_BACKENDS holds all currently running backends for the execution.

The question this answers is "if Linode is one of the backends at all, we use deltas".

File: spread.yaml (hunk 4 above)
Comment (niemeyer) [inline spread.yaml]: Lots of docs already: https://github.com/snapcore/spread#repacking

File: spread.yaml (hunk 2 above)
Comment (niemeyer) [inline spread.yaml]: Note that this magic comes from tar itself (see `--transform`).

**Pull request merged**